nowcasting_dataset/data_sources/satellite/satellite_data_source.py

```diff
@@ -56,11 +59,87 @@ def _get_time_slice(self, t0_dt: pd.Timestamp) -> xr.DataArray:
         start_dt = self._get_start_dt(t0_dt)
         end_dt = self._get_end_dt(t0_dt)
         data = self.data.sel(time=slice(start_dt, end_dt))
-
         assert type(data) == xr.DataArray
 
         return data
 
+    def get_image_pixel(
```

Comment:
I'd suggest changing the name of this function... `get_image_pixel()` sounds like it literally gets a single pixel!

Maybe something like `get_square_crop_of_image()`?  Although maybe that's a big clumsy and verbose! 

Maybe `get_spatial_region_of_interest()`?  (I think I slightly prefer that to `get_square_crop_of_image()` because it's more general...)

And, in general, maybe _every_ `DataSource` should have separate steps to `get_time_slice()` and `get_spatial_region_of_interest()`?  (Which has been on my mind for a while:  See #48!)

Reply:
Done! And yeah. splitting it out I think is a good idea, but maybe leave it to next year?

Reply:
> splitting it out I think is a good idea, but maybe leave it to next year?

definitely :slightly_smiling_face: